lmdeploy/serve/openai/api_server.py

```diff
@@ -871,6 +872,43 @@ def encode(prompt: str, do_preprocess: bool, add_bos: bool):
         return EncodeResponse(input_ids=encoded, length=length)
 
 
+@router.post('/pooling')
+async def pooling(request: PoolingRequest, raw_request: Request = None):
+    """Pooling prompts for reward model.
+
+    In vLLM documentation, https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html#pooling-api_1,
+    the input format of Pooling API is the same as Embeddings API.
+
+    Go to https://platform.openai.com/docs/api-reference/embeddings/create
+    for the Embeddings API specification.
+
+    The request should be a JSON object with the following fields:
+    - model (str): model name. Available from /v1/models.
+    - input (str | List[int]): input text to be embed, encoded as a string or array of tokens
+    """
+
+    async_engine = VariableInterface.async_engine
+
+    request_input = request.input
+    model_name = request.model or async_engine.model_name
+
+    if isinstance(request_input, str):
+        input_ids = async_engine.tokenizer.encode(request_input, add_special_tokens=False)
+    elif isinstance(request_input, List[int]):
+        input_ids = request_input
+    else:
```

Comment:
the input is defined to  accept type of `input: Union[list[int], list[list[int]], str, list[str]]` but here only accept `str, or List[int]`